src/options.cpp

```diff
@@ -364,6 +394,12 @@ void cOpt::setNext()
 
         sSet = vItems[iNext];
 
+    } else if (sType == "string_input") {
+        int iMenuTextLength = sMenuText.length();
+        sSet = string_input_popup("", (iMaxLength > 80) ? 80 : ((iMaxLength < iMenuTextLength) ? iMenuTextLength : iMaxLength+1),
```

Comment:
Had the problem before (v menu filters) with ESC clearing the input but couldn't think of a good way to know that it was canceled so it's just doing that, clearing the input. Any ideas?

I'll update it to check if the string actually changed or not.